src/engine/TextIndexScanForWord.cpp

```diff
@@ -20,9 +20,10 @@ Result TextIndexScanForWord::computeResult(
 
   if (!isPrefix_) {
     IdTable smallIdTable{getExecutionContext()->getAllocator()};
-    smallIdTable.setNumColumns(1);
+    smallIdTable.setNumColumns(2);
     smallIdTable.resize(idTable.numRows());
     std::ranges::copy(idTable.getColumn(0), smallIdTable.getColumn(0).begin());
+    std::ranges::copy(idTable.getColumn(2), smallIdTable.getColumn(1).begin());
```

Comment:
I think we don't need to copy all the columns...
The IdTable class has a function `setColumnSubset` which can be used to do this more efficiently(stripping columns works well with column based layouts).

Reply:
Implemented the function to use setColumnSubset.